src/cpp/shared/combat/army.cpp

```diff
@@ -1965,17 +2229,126 @@ void army::MoveTo(int hexIdx) {
   }
 }
 
+bool army::TargetOnStraightLine(int targHex) {
+  int deltaY = gpCombatManager->combatGrid[targHex].occupyingCreatureBottomY - gpCombatManager->combatGrid[this->occupiedHex].occupyingCreatureBottomY;
+  int deltaX = gpCombatManager->combatGrid[targHex].centerX - gpCombatManager->combatGrid[this->occupiedHex].centerX;
+  double angle = abs((180.0 / 3.141592653589793238463) * atan2(deltaY, abs(deltaX)));
+  return angle == 0 || angle == 62.354024636261322;
+}
+
+int army::GetStraightLineDirection(int targHex) {
+  int deltaY = gpCombatManager->combatGrid[targHex].occupyingCreatureBottomY - gpCombatManager->combatGrid[this->occupiedHex].occupyingCreatureBottomY;
+  int deltaX = gpCombatManager->combatGrid[targHex].centerX - gpCombatManager->combatGrid[this->occupiedHex].centerX;
+  double angle = (180.0 / 3.141592653589793238463) * atan2(deltaY, deltaX);
+  if(angle == -62.354024636261322)
+    return 0;
+  if(angle == 0)
+    return 1;
+  if(angle == 62.354024636261322)
+    return 2;
+  if(angle == 117.64597536373869)
+    return 3;
+  if(angle == 180)
+    return 4;
+  if(angle == -117.64597536373869)
```

Comment:
Testing for equality for floating-point numbers is pretty brittle.  Can you get the behavior you want with something like `(angle > -117.7 && angle < -117.6)`?